precompile/testutils/test_precompile.go

```diff
@@ -69,18 +98,76 @@ func (test PrecompileTest) Run(t *testing.T, module modules.Module, state contra
 		input = test.InputFn(t)
 	}
 
-	if input != nil {
-		ret, remainingGas, err := module.Contract.Run(accesibleState, test.Caller, contractAddress, input, test.SuppliedGas, test.ReadOnly)
-		if len(test.ExpectedErr) != 0 {
-			require.ErrorContains(t, err, test.ExpectedErr)
-		} else {
-			require.NoError(t, err)
-		}
-		require.Equal(t, uint64(0), remainingGas)
-		require.Equal(t, test.ExpectedRes, ret)
+	return PrecompileRunparams{
+		AccessibleState: accesibleState,
+		Caller:          test.Caller,
+		ContractAddress: contractAddress,
+		Input:           input,
+		SuppliedGas:     test.SuppliedGas,
+		ReadOnly:        test.ReadOnly,
+	}
+}
+
+func (test PrecompileTest) Bench(b *testing.B, module modules.Module, state contract.StateDB) {
+	runParams := test.setup(b, module, state)
+
+	if runParams.Input == nil {
+		b.Skip("Skipping precompile benchmark due to nil input (used for configuration tests)")
 	}
 
```

Comment:
It seems like we do `errorcheck`, `benchmark`, and sanity check `errorcheck`? Can we combine the errorcheck inside benchmark? and just do one total combo `errorcheck +benchmark`?

Reply:
https://scene-si.org/2017/06/06/benchmarking-go-programs/

Reply:
I'm not sure I'm following your suggestion, could you be more specific?